test/position-managers/Permit.t.sol

```diff
@@ -0,0 +1,277 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+import {PoolManager} from "@uniswap/v4-core/src/PoolManager.sol";
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {IHooks} from "@uniswap/v4-core/src/interfaces/IHooks.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/src/types/Currency.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/src/types/PoolId.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {StateLibrary} from "@uniswap/v4-core/src/libraries/StateLibrary.sol";
+import {SignatureVerification} from "permit2/src/libraries/SignatureVerification.sol";
+
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+import {IERC721Permit} from "../../src/interfaces/IERC721Permit.sol";
+import {ERC721Permit} from "../../src/base/ERC721Permit.sol";
+import {UnorderedNonce} from "../../src/base/UnorderedNonce.sol";
+
+import {PositionConfig} from "../../src/libraries/PositionConfig.sol";
+import {IPositionManager} from "../../src/interfaces/IPositionManager.sol";
+
+import {PosmTestSetup} from "../shared/PosmTestSetup.sol";
+
+contract PermitTest is Test, PosmTestSetup {
+    using FixedPointMathLib for uint256;
+    using CurrencyLibrary for Currency;
+    using PoolIdLibrary for PoolKey;
+    using StateLibrary for IPoolManager;
+
+    PoolId poolId;
+    address alice;
+    uint256 alicePK;
+    address bob;
+    uint256 bobPK;
+
+    PositionConfig config;
+
+    function setUp() public {
+        (alice, alicePK) = makeAddrAndKey("ALICE");
+        (bob, bobPK) = makeAddrAndKey("BOB");
+
+        deployFreshManagerAndRouters();
+        deployMintAndApprove2Currencies();
+
+        (key, poolId) = initPool(currency0, currency1, IHooks(address(0)), 3000, SQRT_PRICE_1_1, ZERO_BYTES);
+
+        // Requires currency0 and currency1 to be set in base Deployers contract.
+        deployAndApprovePosm(manager);
+
+        seedBalance(alice);
+        seedBalance(bob);
+
+        approvePosmFor(alice);
+        approvePosmFor(bob);
+
+        // define a reusable range
+        config = PositionConfig({poolKey: key, tickLower: -300, tickUpper: 300});
+    }
+
+    function test_domainSeparator() public view {
+        assertEq(
+            IERC721Permit(address(lpm)).DOMAIN_SEPARATOR(),
+            keccak256(
+                abi.encode(
+                    keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"),
+                    keccak256("Uniswap V4 Positions NFT"), // storage is private on EIP712.sol so we need to hardcode these
+                    keccak256("1"),
+                    block.chainid,
+                    address(lpm)
+                )
+            )
+        );
+    }
+
+    function test_permit_increaseLiquidity() public {
```

Comment:
you dont actually need permissions to increase, so can probably remove this?

Reply:
or you can leave it to show you dont need a permit

Reply:
removing

the test `test_increaseLiquidity_withUnapprovedCaller` shows that users can increase without approval